File: GOVI-Back/src/main/java/com/sncf/govi/service/model/Mission.java
package com.sncf.govi.service.model;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Builder
@Data
public class Mission {
    private String codeMission;
    private String gareDepart;
    private String gareArrivee;
    private String heureArrivee;
    private String heureDepart;
    private TransporteurEnum transporteurEnum;
    private CouleurEnum couleurEnum;
    private Conducteur conducteurTrain;
    @Builder.Default
    private List<Conducteur> conducteursEVLoc = new ArrayList<>();
}

File: GOVI-Front/src/app/model/Mission.ts
import { Conducteur } from './Conducteur';

export class Mission {
  codeMission: string;
  gareDepart: string;
  gareArrivee: string;
  heureArrivee: Date;
  heureDepart: Date;
  couleurEnum: string;
  conducteurTrain: Conducteur;
  conducteursEVLoc: Conducteur[];

  constructor(
    codeMission: string,
    gareDepart: string,
    gareArrivee: string,
    heureArrivee: Date,
    heureDepart: Date,
    couleurEnum: string,
    conducteurTrain: Conducteur,
    conducteursEVLoc: Conducteur[]
  ) {
    this.codeMission = codeMission;
    this.gareDepart = gareDepart;
    this.gareArrivee = gareArrivee;
    this.heureArrivee = heureArrivee;
    this.heureDepart = heureDepart;
    this.couleurEnum = couleurEnum;
    this.conducteurTrain = conducteurTrain;
    this.conducteursEVLoc = conducteursEVLoc;
  }
}
